# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1503519480 -7200
#      Wed Aug 23 22:18:00 2017 +0200
# Node ID 4be7a254f7521e308c9f7dc1ac09c953a6300e47
# Parent  ab5780d548693fcefd245fe262756ec01afe351e
fix some icon issues of the variable edtor

* variable-editor.cc (construct_tool_bar): insert correct icon for the pull down menu;
  (notice_settings): read and set the icon size from the settings file

diff --git a/libgui/src/variable-editor.cc b/libgui/src/variable-editor.cc
--- a/libgui/src/variable-editor.cc
+++ b/libgui/src/variable-editor.cc
@@ -127,17 +127,17 @@ variable_editor::construct_tool_bar ()
      this, SLOT (pasteTableClipboard ()));
   tool_bar->addSeparator ();
 
   //QAction *print_action; /icons/fileprint.png
   //tool_bar->addSeparator ();
 
   QToolButton *plot_tool_button = new QToolButton (tool_bar);
   plot_tool_button->setText (tr ("Plot"));
-  plot_tool_button->setIcon (resource_manager::icon("logo"));//QIcon (":/actions/icons/gear.png")); // TODO
+  plot_tool_button->setIcon (resource_manager::icon ("applications-system"));
 
   plot_tool_button->setPopupMode (QToolButton::InstantPopup);
 
   QMenu *plot_menu = new QMenu (tr ("Plot"), plot_tool_button);
   plot_menu->setSeparatorsCollapsible (false);
   QSignalMapper *plot_mapper = new QSignalMapper (plot_menu);
   plot_mapper->setMapping(plot_menu->addAction ("plot",
                                                 plot_mapper,
@@ -1106,16 +1106,28 @@ void variable_editor::notice_settings(co
                                               + class_chars.mid (i,1),
                                               default_var).value<QColor> ();
       table_colors.replace (i,setting_color);
     }
   update_colors();
   int toolsize = settings->value("variable_editor/toolbar_size",QVariant(0)).toInt();
   if (toolsize > 0)
       tool_bar->setIconSize(QSize(toolsize,toolsize));
+
+  // Icon size in the toolbar
+  int icon_size_settings = settings->value ("toolbar_icon_size",0).toInt ();
+  QStyle *st = style ();
+  int icon_size = st->pixelMetric (QStyle::PM_ToolBarIconSize);
+
+  if (icon_size_settings == 1)
+    icon_size = st->pixelMetric (QStyle::PM_LargeIconSize);
+  else if (icon_size_settings == -1)
+    icon_size = st->pixelMetric (QStyle::PM_SmallIconSize);
+
+  tool_bar->setIconSize (QSize (icon_size,icon_size));
 }
 
 /// Also updates the font
 void variable_editor::update_colors()
 {
   stylesheet="";
   stylesheet += "QTableView::item{ foreground-color: " + table_colors[0].name() +" }";
   stylesheet += "QTableView::item{ background-color: " + table_colors[1].name() +" }";
